File: Domain/Entities/LoanCalculation.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain.Entities
{
    public class LoanCalculation
    {
        public decimal Amount { get; set; }
        public double Interest { get; set; }
        public int Years { get; set; }
    }
}

File: frontend/src/app/models/loanCalculation.ts
export interface LoanCalculation {
  amount: number;
  interest?: number;
  years: number;
}

export class LoanCalculationFormValues {
  amount: number = 0;
  interest?: number = 3.5;
  years: number = 0;

  constructor(loanCalculation?: LoanCalculationFormValues) {
    if (loanCalculation) {
      this.amount = loanCalculation.amount;
      this.interest = loanCalculation.interest;
      this.years = loanCalculation.years;
    }
  }
}